test-data/unit/fine-grained.test

```diff
@@ -3886,6 +3886,173 @@ main:3: error: Value of type variable "submod.T" of "D" cannot be "S"
 main:3: error: Value of type variable "submod.U" of "D" cannot be "I"
 ==
 
+[case testGenericFineCallableNormal]
+import a
+[file a.py]
+import b
+x: int = b.f(int())
+[file b.py]
+from c import g
+f = g
+[file c.py]
+from typing import TypeVar
+class B: pass
+T = TypeVar('T')
+def g(x: T) -> T:
+    pass
+[file c.py.2]
+from typing import TypeVar
+class B: pass
+T = TypeVar('T', str, B)
+def g(x: T) -> T:
+    pass
+[out]
+==
+a.py:2: error: Value of type variable "T" of function cannot be "int"
+
+[case testGenericFineCallableNamed]
+import a
+[file a.py]
+import b
+x: int = b.f(x=int())
+[file b.py]
+from c import g
+f = g
+[file c.py]
+from typing import TypeVar
+class B: pass
+T = TypeVar('T')
+def g(x: T) -> T:
+    pass
+[file c.py.2]
+from typing import TypeVar
+class B: pass
+T = TypeVar('T')
+def g(y: T) -> T:
+    pass
+[out]
+==
+a.py:2: error: Unexpected keyword argument "x"
+c.py:4: note: called function defined here
+
+[case testGenericFineCallableInBound]
+import a
+[file a.py]
+import b
+x: int = b.f()(int())
+[file b.py]
+from c import g
+f = g
+[file c.py]
+from typing import Callable, TypeVar
+class B: pass
+T = TypeVar('T')
+def g() -> Callable[[T], T]:
+    pass
+[file c.py.2]
+from typing import Callable, TypeVar
+class B: pass
+T = TypeVar('T', str, B)
+def g() -> Callable[[T], T]:
+    pass
+[out]
+==
+a.py:2: error: Value of type variable "T" of function cannot be "int"
+
+[case testGenericFineCallableAddedBound]
+import a
+[file a.py]
+import b
+x: int = b.f(int())
+[file b.py]
+from c import g
+f = g
+[file c.py]
+from typing import TypeVar
+class B: pass
+T = TypeVar('T')
+def g(x: T) -> T:
+    pass
+[file c.py.2]
+from typing import TypeVar
+class B: pass
+T = TypeVar('T', bound=B)
+def g(x: T) -> T:
+    pass
+[out]
+==
+a.py:2: error: Value of type variable "T" of function cannot be "int"
+
+[case testGenericFineCallableBoundDeleted-skip-nocache]
```

Comment:
You can create a separate issue about this failure and we look at it later.

Reply:
OK, will create a separate issue.